dev-tools/timing/Types.hs

```diff
@@ -0,0 +1,272 @@
+{-# LANGUAGE DeriveFunctor #-}
+
+module Types (module Types) where
+
+import Data.Aeson (FromJSON (..), Value (..))
+import Data.Aeson.Key (toString)
+import Data.Aeson.KeyMap (toList)
+import Data.HashMap.Strict qualified as HashMap
+import Data.Map (Map)
+import Data.Map qualified as Map
+import Data.Maybe (catMaybes, fromMaybe)
+import Data.Sequence (Seq (..))
+import Data.Set qualified as Set
+import Data.Text (Text, pack)
+import Data.Text qualified as Text
+import Data.Vector qualified as Vector
+import GHC.Generics (Generic)
+import Profiteur.Core
+
+data Context = Plain Text | Ref Text Text deriving (Eq, Ord, Show)
+
+instance FromJSON Context where
+    parseJSON (String t) = pure $ Plain t
+    parseJSON (Object o) | [(k, String v)] <- toList o = pure $ Ref (pack $ toString k) v
```

Comment:
Maybe [use `toText`?](https://hackage.haskell.org/package/aeson-2.2.3.0/docs/src/Data.Aeson.Key.html#toText) 
```suggestion
    parseJSON (Object o) | [(k, String v)] <- toList o = pure $ Ref (toText k) v
```
if you don't want to use the new context type